src/event-mapper.ts

```diff
@@ -29,9 +29,22 @@ export function eventMapperFor(client: MatrixClient, options: MapperOpts): Event
     const decrypt = options.decrypt !== false;
 
     function mapper(plainOldJsObject: Partial<IEvent>) {
```

Comment:
I think there would be value in testing that we're not creating dupes. 
Especially given some of the nasty that can bring

Reply:
Good call, will do so shortly